src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -488,15 +489,48 @@ private static string GetClaimValueType(object obj)
             return objType.ToString();
         }
 
+        /// <summary>
+        /// Gets a <see cref="Claim"/> representing the { key, 'value' } pair corresponding to the provided <paramref name="key"/>.
+        /// </summary>
+        /// <remarks>If the key has no corresponding value, this method will throw.</remarks>   
+        public Claim GetClaim(string key)
+        {
+            string issuer = Issuer ?? ClaimsIdentity.DefaultIssuer;
+
+            if (!Payload.TryGetValue(key, out var jTokenValue))
+                throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14304, key)));
+
+            if (jTokenValue == null)
```

Comment:
I'm not sure that this will ever be null.
Perhaps we should check for null with `if (jTokenValue.Type == JTokenType.Null)`.

There is no test where the claim below will returned:
`return new Claim(key, string.Empty, JsonClaimValueTypes.JsonNull, issuer, issuer);`

Reply:
@GeoK You're right!

I wrote the tests for this incorrectly; they should have been checking to make sure that the value of the 'nill' claim was an empty string, but instead the expected value was "null." 

This was the value my tests were checking against (which is incorrect):
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/fbc717d431cd90e51eb50e325e513a512601a6a4/test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs#L54

I am wondering though, does it make sense to return an empty string if the value is null?

Reply:
@mafurman - I don't think that it really matters as long as the type is `JsonClaimValueTypes.JsonNull`. 
null might be a better option, but a Claim can't be created with a value set as null.

Reply:
@mafurman would it make sense to have the value == string.empty when the type is JsonClaimValueTypes.JsonNull?

Reply:
@brentschmaltz I think that having value == string.Empty is the best option considering that's consistent with the behavior of JwtSecurityToken.

Reply:
👍 :shipit: